shlomobot_pytest/common_tests.py

```diff
@@ -79,63 +84,57 @@ def find_functions_with_single_quote_docstrings(file_list: list[str]) -> list[st
     return single_quote_docstrings
 
 
-def contains_main_function(module_name: str) -> bool:
+def contains_main_function(py_filename: str) -> bool:
     """
     Checks if the 'main' function exists within module
     """
-    module = import_pyfile(module_name)
+    module = import_pyfile(py_filename)
     try:
         callable(getattr(module, "main"))
         return True
     except AttributeError:
         return False
 
 
-def builtins_not_used_as_variable(file_list: list[str]) -> bool:
+def builtins_not_used_as_variable(function: FunctionType) -> bool:
     """
-    Checks if inside the files given in 'file_list' any builtins are used as variables
-
-    Return True if no builtins are used as variables, False if at least one is.
+    Return True if no builtins are used as variables in the function, else return True
     """
-    builtins_list = [word for word in dir(builtins) if not re.match("[A-Z|_]", word[0])]
-    functions_list = get_functions_from_files(file_list)
-
     builtin_used_as_variable = r"\n\s*(?:[^\s]*? ?, ?)*?{0} ?(?:\s*,\s*[^\W]+?\s*)*=.*"
     builtin_given_to_function = r"def {0}\((?:[^\s]*? ?, ?)*?{1}(?:\s*,\s*[^\W]+?)*\):"
+    builtins_list = [word for word in dir(builtins) if not re.match("[A-Z|_]", word[0])]
 
-    for function_name, function in functions_list:
-        # extract function code
-        function_code = inspect.getsource(function)
-        # look for all possible builtin words used in the function
-        for builtin_word in builtins_list:
-            builtin_as_variable_regex = builtin_used_as_variable.format(builtin_word)
-            builtin_as_paramater_regex = builtin_given_to_function.format(
-                function_name, builtin_word
-            )
-            occurences_as_variable = re.search(builtin_as_variable_regex, function_code)
-            occurences_as_paramater = re.search(
-                builtin_as_paramater_regex, function_code
-            )
-            if occurences_as_variable or occurences_as_paramater:
-                return False
+    function_name = function.__name__
+
+    # look for all possible builtin words used in the function
+    for builtin_word in builtins_list:
+        builtin_variable_regex = builtin_used_as_variable.format(builtin_word)
+        builtin_paramater_regex = builtin_given_to_function.format(
+            function_name, builtin_word
+        )
+
+        if function_contains_regex(
+            builtin_variable_regex, function
+        ) or function_contains_regex(builtin_paramater_regex, function):
```

Comment:
How about we just combine the two regexes?

Perhaps something like this can work?
```suggestion
        builtin_parameter_or_variable_pattern = f"({builtin_variable_regex})|({builtin_variable_regex})"
        if function_contains_regex(builtin_parameter_or_variable_pattern, function):
```